cmd/krew/cmd/update.go

```diff
@@ -41,12 +57,147 @@ Remarks:
 	RunE: ensureIndexUpdated,
 }
 
+func retrieveUpdatedPluginList() ([]string, error) {
+	modifiedFiles, err := gitutil.ListModifiedFiles(constants.IndexURI, paths.IndexPluginsPath())
+	if err != nil {
+		return []string{}, err
+	}
+
+	var plugins []string
+	for _, f := range modifiedFiles {
+		filename := filepath.Base(f)
+		extension := filepath.Ext(filename)
+		name := filename[0 : len(filename)-len(extension)]
+		plugins = append(plugins, name)
+	}
+
+	return plugins, nil
+}
+
+func retrievePluginNameVersionMap(names []string) map[string]string {
+	m := make(map[string]string, len(names))
+	for _, n := range names {
+		plugin, err := indexscanner.LoadPluginByName(paths.IndexPluginsPath(), n)
+		if err != nil {
```

Comment:
You should only continue for `os.IsNoExist(err)`. All other errors should abort the function.